### PR TITLE
Ports dynamic gamemode from /vg/ + also normalizes head/antag restrictions

### PR DESCRIPTION
**Pull request by Sishen1542**
## About The Pull Request
port of https://github.com/tgstation/tgstation/pull/44639

Port the dynamic gamemode from /vg/.
(Really bad explanation of the mode incoming.)
The dynamic game mode generates a threat number which is used to "buy" rulesets (rulesets are basically your antagonists). This means you can have rounds with for example traitors and cult (you can have up to three roundstart rulesets depending on the pop and threat level), and then there are latejoin and midround rulesets which basically do what they say (latejoin ruleset assigns late joining player as an antagonists and midround assigns ghosts or a currently alive player as an antagonist)

also makes it so heads can roll less antags

## Why It's Good For The Game
This increases the chances of people getting their important antagonist role and makes rounds more interesting (when cultists gets their hand on wizard's magic) when everything can happen at the same time (cult, wiz and traitor could happen on high threat level).

## Changelog
:cl: original by TheChosenEvilOne, port by sishen1542
add: Ported dynamic mode from /vg/, originally made by @DeityLink, @Kurfursten and @ShiftyRail
/:cl:

### CODE REVIEW

File: tgstation.dme [hunk 1]
```diff
@@ -2926,6 +2931,7 @@
 #include "modular_citadel\code\modules\arousal\toys\dildos.dm"
 #include "modular_citadel\code\modules\awaymissions\citadel_ghostrole_spawners.dm"
 #include "modular_citadel\code\modules\cargo\console.dm"
+#include "modular_citadel\code\modules\cargo\packs.dm"
```
Comment (deathride58) [inline tgstation.dme]: The contents of this file were demodularized. Remove it from the DME

File: tgstation.dme [hunk 2]
```diff
@@ -2959,6 +2965,7 @@
 #include "modular_citadel\code\modules\clothing\under\trek_under.dm"
 #include "modular_citadel\code\modules\clothing\under\turtlenecks.dm"
 #include "modular_citadel\code\modules\clothing\under\under.dm"
+#include "modular_citadel\code\modules\crafting\recipes.dm"
```
Comment (deathride58) [inline tgstation.dme]: Ditto

File: tgstation.dme [hunk 3]
```diff
@@ -2983,6 +2990,7 @@
 #include "modular_citadel\code\modules\mentor\mentorhelp.dm"
 #include "modular_citadel\code\modules\mentor\mentorpm.dm"
 #include "modular_citadel\code\modules\mentor\mentorsay.dm"
+#include "modular_citadel\code\modules\mining\mine_items.dm"
```
Comment (deathride58) [inline tgstation.dme]: Ditto

File: tgstation.dme [hunk 4]
```diff
@@ -3040,6 +3048,7 @@
 #include "modular_citadel\code\modules\research\designs\xenobio_designs.dm"
 #include "modular_citadel\code\modules\research\designs\weapon_designs\weapon_designs.dm"
 #include "modular_citadel\code\modules\research\techweb\_techweb.dm"
+#include "modular_citadel\code\modules\research\techweb\all_nodes.dm"
```
Comment (deathride58) [inline tgstation.dme]: Ditto

**Comment by RogueSteampunker** [comment]
The idea itself is interesting, and can certainly make antags a great deal more interesting and the like. However, trying to use it as a whole-sale replacement of the default vote, especially if extended is going to be weighed to only proc when lowpop is present and few people vote for anything besides low threat, seems like a great way to alienate a fair deal of the base who isn't explicitly looking for 24/7 antag, especially when we already have a major issue of validhunters trying to force secret as much as they can.

**Comment by DeityLink** [comment]
Oh god another one.

**Comment by nik707** [comment]
im afraid to touch this ask the other two

**Review by BlackMajor — APPROVED** [review]
Looks good to me but good golly oh hell we're gonna have to get some balances going for this later down the line.

Should be fun.

**Pull request merged**